direct-messages/README.md

```diff
@@ -10,24 +10,74 @@ because the same message can simultaneously:
 - give the new person the info needed to access the group
 - reveal to the group that this action happened
 
-## Mapping `feed_id` to a `recp_key`
+## Computing the `recp_key`
 
 We define a shared key that the sender + recipient can both derive:
 
 ```js
-function computeDirectMessageKey (my_secret, your_public) {
+function computeDirectMessageKey (my_dh_secret, my_dh_public, your_dh_public, my_feed_tfk, your_feed_tfk) {
   var hash = 'SHA256'
-  var salt = SHA256("envelope-direct-messsage-shared-key-extract-salt")
-  var input_keying_material = scalarmult(my_secret, your_public)
-
-  return hkdf.Extract(hash, salt, input_keying_material)
+  var length = 32
+  var salt = SHA256("envelope-dm-v1-extract-salt")
+  var info = slp_encode(
+      ["envelope-ssb-dm-v1/key"]
+      .concat(sort([
+		tfk_encode(my_dh_public) || my_feed_tfk,
+		tfk_encode(your_dh_public) || your_feed_tfk
+  ])))
+  var input_keying_material = scalarmult(my_dh_secret, your_dh_public)
+
+  return hkdf(input_keying_material, length, {salt, info, hash})
 }
 ```
 
-Notes:
-- we (curently) use the primary feed keys for this derivation
-- for feeds based on `ed25519` keypairs, we convert these to `curve25519` keypairs before doing scalarmult
-
+The function arguments are:
+- `my_dh_secret`: the own Diffie-Hellman secret key.
+- `my_dh_public`: the own Diffie-Hellman public key.
+- `your_dh_public`: the Diffie-Hellman public key of the communication partner.
+- `my_feed_tfk`: the own Scuttlebutt feed ID in tfk format, as specified in TODO.
+- `your_feed_tfk`: the Scuttlebutt feed ID in tfk format of the communication partner.
+
+The constants we use are:
+- `hash` is set to `SHA256`, such that HKDF uses SHA256 as underlying hash function.
+- `length` is set to 32, such that the output keys are 32 bytes = 256 bits long.
+- `salt` is set to `SHA256("envelope-dm-v1-extract-salt")`.
```

Comment:
are you saying you want the salt to be the SHA256 hash of that String?
My guess is that the hash there is unnecessary - we just need the salt to be a unique string to this context yes? mapping one unique string deterministically to another unique string doesn't add anything does it ... unless that sha256 outputs a longer string ...

Reply:
also, we might like to say all inputs (other than `hash` and `length`) are Buffers.

Reply:
As a general rule, a salt should be a "non-secret random string" (from the HKDF RFC 5869). While "the hash of a descriptive string" is not strictly random, it still comes pretty close.

We should probably compute the hash before publishing the spec and then hard-code it in the implementations.